### PR TITLE
change to cobra, Improved code

### PR DESCRIPTION
**Pull request by aimuz**
Signed-off-by: aimuz <mr.imuz@gmail.com>

### CODE REVIEW

**Comment by bwplotka** [comment]
Thanks for this contribution! Unfortunately, I can't accept it. I don't see a reason for changing the framework, plus the same "code improvements" can be done with the existing framework we are familiar with (even though it's not supported anymore).

Next time, please make sure to discuss such idea before submitting the code. Plus some description on why it should be merge would be nice! 🤗 Thanks! Closing for now, happy to reopen if you disagree with me.

**Comment by aimuz** [comment]
I think the existing command prompt is too confusing, it may be clear for the maintainer, but for the user, he needs a good command prompt. The refactoring will make it simpler to use and I think it will be easier for contributors to understand in terms of code

File: README.md [hunk 1]
```diff
@@ -258,43 +258,27 @@ See an extensive and up-to-date description of the `bingo` usage below:
...
-Prints bingo Version.
+Bingo is a command line tool for Like `go get`  but for Go tools! 
+CI Automating versioning of Go binaries in a nested, isolated Go modules.
+
+Usage:
+  bingo [command]
+
+Available Commands:
+  completion  Generate the autocompletion script for the specified shell
+  get
```
Comment (bwplotka) [inline README.md:269]: Any reason why get - most important command does not have any help? 🤔

File: main.go [hunk 2]
```diff
@@ -6,224 +6,43 @@
...
-			logger.Fatalf("Error: %+v", errors.Wrapf(err, "%s command failed", flags.Arg(0)))
-		}
-		logger.Fatalf("Error: %v", errors.Wrapf(err, "%s command failed", flags.Arg(0)))
+	rootCmd := NewBingoCommand(logger)
+	err := rootCmd.Execute()
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
 	}
+	return
```
Comment (bwplotka) [inline main.go:45]: nit: Do we need this? (no need for return in main)

File: main.go [hunk 3]
```diff
@@ -6,224 +6,43 @@
...
-		cmdFunc = func(ctx context.Context, r *runner.Runner) error {
-			_, err := fmt.Fprintln(os.Stdout, version.Version)
-			return err
-		}
-	default:
-		exitOnUsageError(flags.Usage, "No such command", flags.Arg(0))
-	}
-
-	g := &run.Group{}
-	g.Add(run.SignalHandler(context.Background(), syscall.SIGINT, syscall.SIGTERM))
```
Comment (bwplotka) [inline main.go:201]: This allowed us to Ctrl+C and gracefully finish our CLI, I wonder if it works well now.

**Review by bwplotka — COMMENTED** [review]
Looks quite good. One question so far, otherwise looks quite good. I would need to manually play with it to check if there is any surprising behaviour with flags or interrupt

**Comment by bwplotka** [comment]
Do you mind rebasing?

File: main.go (hunk 3 above)
Comment (aimuz) [reply main.go:201]: It seems that I have omitted the logic of this part

File: main.go (hunk 2 above)
Comment (aimuz) [reply main.go:45]: Yes, I'll delete him

**Comment by aimuz** [comment]
Fixed

File: README.md (hunk 1 above)
Comment (aimuz) [reply README.md:269]: Fixed

**Comment by bwplotka** [comment]
Thanks for fixing tests - approval for CI is annoying - we will get rid of this once you submit your first contribution 💪🏽 Thanks for patience!

File: README.md [hunk 4]
```diff
@@ -258,43 +258,28 @@ See an extensive and up-to-date description of the `bingo` usage below:
 ## Command Help
 
 ```bash mdox-exec="bingo --help" mdox-expect-exit-code=2
-bingo: 'go get' like, simple CLI that allows automated versioning of Go package level binaries (e.g required as dev tools by your project!)
-built on top of Go Modules, allowing reproducible dev environments. 'bingo' allows to easily maintain a separate, nested Go Module for each binary.
-
+Bingo is a command line tool for Like `go get`  but for Go tools! 
+CI Automating versioning of Go binaries in a nested, isolated Go modules.
```
Comment (bwplotka) [inline README.md:262]: I wished we could work on description in separate PR e.g I don't get "CI Automatic versioning.." sentence.

File: README.md [hunk 5]
```diff
@@ -258,43 +258,28 @@ See an extensive and up-to-date description of the `bingo` usage below:
...
-
-  -moddir string
-    	Directory where separate modules for each binary is maintained. If does not exists, bingo list will fail. (default ".bingo")
-  -v	Print more'
+Usage:
+  bingo [command]
 
+Available Commands:
+  completion  Generate the autocompletion script for the specified shell
+  get         go get like, simple CLI that allows automated versioning of Go package level
```
Comment (bwplotka) [inline README.md:270]: Hm.. still, example would be nice to have somehow. We could merge and fix later as well.

**Review by bwplotka — APPROVED** [review]
The descriptions, could be improved, but overall not bad, good to go. Thanks! 

We can fix wording later before next release. 💪🏽

**Pull request merged**